app/controllers/application_controller.rb

```diff
@@ -22,6 +22,39 @@ class ApplicationController < ActionController::API
 
   private
 
+  set_current_tenant_through_filter
+  before_action :set_the_current_tenant
+  after_action  :unset_the_current_tenant
+
+  def set_the_current_tenant
+    return unless Tenant.tenancy_enabled?
```

Comment:
If tenancy is disabled, then doesn't the sources_controller end up creating a tenant anyway since it executes before this line?

Reply:
Yes, but I didn't think it would hurt to let it do that